.github/workflows/translation.yml

```diff
@@ -0,0 +1,275 @@
+            --verbose 2> >(grep -i Operation-Location) >/dev/null | grep -oEi '(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}')
+          echo opertaion id is $operation_id
+          echo "operation_id=$operation_id" >> $GITHUB_ENV
+      ############################
+      # Wait for Translation     #
+      ############################
+      - name: "Wait for translation job"
+        run: |
+          set -eux
+          action_state="${{ env.action_state }}"
+          echo $action_state
+          status=NotStarted
+          while [ "$status" = "NotStarted" ]
+          do
+            status=$(az rest --method get \
+              --url ${{ secrets.TRANSLATOR_API_URI }}translator/text/batch/v1.0/batches/$operation_id \
+              --headers Ocp-Apim-Subscription-Key=${{ secrets.TRANSLATOR_API_KEY }} | jq -r .status)
+            echo waiting for run to start status is $status
+          done
+
+          while [ "$status" = "Running" ]
+          do
+            status=$(az rest --method get \
+              --url ${{ secrets.TRANSLATOR_API_URI }}translator/text/batch/v1.0/batches/$operation_id \
+              --headers Ocp-Apim-Subscription-Key=${{ secrets.TRANSLATOR_API_KEY }} | jq -r .status)
+            echo running status is $status
+          done
+          echo status is $status
+          if [ "$status" != "Succeeded" ]
+          then
+            echo "Error!" 1>&2
+            exit 1
+          fi
+      #############################
+      # Download translation from #
+      #############################
+      - uses: actions/checkout@v3
+      - name: "Download translation with Azure CLI"
+        run: |
+          set -eux
+          mkdir -p ./translations/${{ matrix.language-id }}
+          end=`date -u -d "30 minutes" '+%Y-%m-%dT%H:%MZ'`
+          sas=`az storage container generate-sas -n $OUTPUT_CONTAINER_NAME --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} --https-only --permissions lr --expiry $end -o tsv`
+          az storage blob download \
+          --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} \
+          --container-name $OUTPUT_CONTAINER_NAME \
+          --name  ${{ vars.DOCUMENT_PATH }} \
+          --file  ./translations/${{ matrix.language-id }}/${{ vars.DOCUMENT_PATH }} \
+          --sas-token $sas
+      #############################
+      # Delete output container   #
+      #############################
+      - name: Delete target container
+        run: |
+          set -eux
+          az storage container delete \
+            --name $OUTPUT_CONTAINER_NAME \
+            --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} \
+            --auth-mode login
+      #############################
+      # Create PR for language    #
+      #############################
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          branch: feature/${{ matrix.language }}-translation
```

Comment:
when do you push to the branch?

Reply:
the x-translation branch is updated if it wasn't pushed to main.